students/Chris_Kenyon/Lesson_3/Mailroom_Pt1.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python3
+
+# Lesson_3 Activity 4 Mailroom Part 1
+
+
+def page_break():
+    """ Print a separator to distinguish new 'pages'"""
+    print("_"*75+"\n")
+
+
+def menu_page():
+    """ Main Menu Page """
+    while True:
+        try:
+            print("Please choose one of the following options(1,2,3):"
+                  "\n1. Send a Thank you. \n2. Create a report\n3. Quit")
+            option = int(input('--->'))
+        except ValueError:
+            print("you have made an invalid choice, try again.")
+            page_break()
+            continue
+        if option < 1 or option > 3:
+            print("you have made an invalid choice, try again.")
+            page_break()
+            continue
+        return option
+
+
+def send_thanks():
+    """ Send Thanks """
+    page_break()
+    while True:
+        list_names = [item[0] for item in donor_chart]
+        amount = []
+        try:
+            print("To whom would you like to say thank you?\n"
+                  "(type \"list\" for a full list of names or"
+                  "\"exit\" to return to the menu)")
+            name = input("--->")
+        except ValueError:
+            print("you have made an invalid choice, try again.")
+            page_break()
+            continue
+        if name == 'list':
+            print(("{}\n"*len(list_names)).format(*list_names))
+            continue
+        elif name in list_names:
+            amount = getAmount()
+            print(amount)
+            if amount == 'exit':
+                break
+            else:
+                donor_chart[list_names.index(name)][1].append(amount)
+        elif name.lower() == 'exit':
+            break
+        else:
+            addname = input("The name you selected is not in the list,"
+                            "would you like to add it(y/n)? ")
+            if addname[0].lower() == 'y':
+                amount = getAmount()
+                print(amount)
+                if amount == 'exit':
+                    break
+                else:
+                    donor_chart.append([name, [amount]])
+            elif addname.lower() == 'exit':
+                break
+            else:
+                print("\nName was not added, try again\n")
+                continue
+        print("\nDear {} \nThank you for your generous donation of ${:.2f}!!\n"
+              "Now all of the kittens will get "
+              "to eat this year".format(name, amount))
+        break
+    return
+
+
+def getAmount():
+    while True:
+        try:
+            amount = input("How much did they donate: ")
+            if str(amount).lower() == 'exit':
+                return amount.lower()
+            else:
+                return float(amount)
+        except:
```

Comment:
It's always better to narrow in on a specific exception, rather than catching all exceptions with a bare `except` statement, because sometimes there can be errors that you don't anticipate and you can don't want those to pass silently. This would be a good place to catch a `ValueError`, as that will be raised by a failed `float()` conversion for non-numeric types.